app/interactions/assignment_interactions/base_assignment_interaction.rb

```diff
@@ -34,5 +35,21 @@ def self.run(*args)
     end
 
     private_class_method :new
+
+    private
+
+    # Subtract the assignment's reward value from the member's rewards_balance
+    # and set the assignment's reward_applied value to false
+    def rollback_rewards
```

Comment:
I think these 2 methods would be best done inside a transaction. If the first update call succeeds and the 2nd fails, you've added/subtracted points but have not updated the reward_applied properly, perhaps leading to eventual improper re-balancing of the reward points